api/src/db/models/user_models.py

```diff
@@ -162,3 +167,21 @@ class UserOpportunityNotificationLog(ApiSchemaTable, TimestampMixin):
     opportunity: Mapped[Opportunity] = relationship(
         "Opportunity", back_populates="all_opportunity_notification_logs"
     )
+
+
+class ApplicationUser(ApiSchemaTable, TimestampMixin):
+    """Link table between User and Application"""
+
+    __tablename__ = "application_user"
+
+    application_id: Mapped[uuid.UUID] = mapped_column(
+        UUID,
+        ForeignKey("api.application.application_id"),
+        primary_key=True,
+    )
+    user_id: Mapped[uuid.UUID] = mapped_column(
+        UUID, ForeignKey("api.user.user_id"), primary_key=True
+    )
+
+    application: Mapped["Application"] = relationship("Application", back_populates="users")
+    user: Mapped["User"] = relationship("User", back_populates="applications")
```

Comment:
The types for these are defined above/imported, can we do?
```suggestion
    application: Mapped[Application] = relationship(Application, back_populates="users")
    user: Mapped[User] = relationship(User, back_populates="applications")
```